includes/Modules/Analytics_4.php

```diff
@@ -945,6 +961,53 @@ protected function create_data_request( Data_Request $data ) {
 						$property_id,
 						$post_body
 					);
+			case 'GET:audience-settings':
+				return function() {
+					return $this->audience_settings->get();
+				};
+			case 'POST:audience-settings':
+				$settings = $data['settings'];
+				if ( ! isset( $settings['configuredAudiences'] ) ) {
+					return new WP_Error(
+						'missing_required_param',
+						/* translators: %s: Missing parameter name */
+						sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'configuredAudiences' ),
+						array( 'status' => 400 )
+					);
+				}
+
+				if ( ! is_array( $settings['configuredAudiences'] ) ) {
+					return new WP_Error(
+						'rest_invalid_param',
+						/* translators: %s: Invalid parameter */
+						sprintf( __( 'Invalid parameter: %s.', 'google-site-kit' ), 'configuredAudiences' ),
+						array( 'status' => 400 )
+					);
```

Comment:
We should probably create a class similar to `Missing_Required_Param_Exception` ie. `Invalid_Param_Exception` to handle this as well.

Reply:
Created a new `Invalid_Param_Exception` class.